### ISSUE TITLE
Add downPath and pathOf

### ISSUE THREAD

**Issue by gabejohnson** [polytypic/fastener#3]
Ramda has `lensPath`, a convenience function for making a lens from a list of keys. fastener could benefit from a similar function which returns a zipper focused at `path`:
```js
seq(F.toZipper({xs: [1, {y: 2}, 3], z: 0}), F.downPath(['xs', 1, 'y']))
// { left: null,
//   focus: 2,
//   key: 'y',
//   right: null,
//   up:
//    { left: [null, 1],
//      key: 1,
//      right: [null, 3],
//      up:
//       { left: null,
//         key: 'xs',
//         right: [null, 0, 'z']}}}
```

this is equivalent to:
```js
seq(F.toZipper({xs: [1, {y: 2}, 3], z: 0}),
    F.downTo('xs'),
    F.downTo(1),
    F.downTo('y')))
```
or
```js
seq(F.toZipper({xs: [1, {y: 2}, 3], z: 0}),
    F.downTo('xs'),
    F.downHead,
    F.right,
    F.downTo('y')))
```

It would also be interesting to extract the path from a zipper:
```js
seq(F.toZipper({xs: [1, {y: 2}, 3], z: 0}),
    F.downPath(['xs', 1, 'y']),
    F.pathOf)
// ['xs', 1, 'y']
```

Another possibility is `zipperPath`:
```js
seq({xs: [1, {y: 2}, 3], z: 0}, F.zipperPath(['xs', 1, 'y']))
```
equivalent to:
```js
seq(F.toZipper({xs: [1, {y: 2}, 3], z: 0}), F.downPath(['xs', 1, 'y']))
```

**Comment by polytypic** [comment]
Yes, `downPath` and `pathOf` seem like good additions.

The current API of this library is fairly minimal and it could be useful to enrich it. There are also some primitive operations (if one considers the zipper format as abstract) missing, such as `insertBefore` and `insertAfter` for inserting elements before or after the focus when inside an object or array.

Regarding `zipperPath`, note that you can write:

```js
seq({xs: [1, {y: 2}, 3], z: 0}, F.toZipper, F.downPath(['xs', 1, 'y']))
```

I'll comment further in the [PR](https://github.com/polytypic/fastener/pull/4).

**Comment by polytypic** [comment]
Released `2.1.0` with `downPath` and `pathOf`.  Thanks again for the PR!  Speaking of `lensPath`, you might find my [`partial.lenses`](https://github.com/calmm-js/partial.lenses/) library interesting.  It is a comprehensive, high-performance optic library.
